File: api/enums/languages.py
import graphene


class LanguageEnums(graphene.Enum):
    ENGLISH = "english"
    FRENCH = "french"

    @property
    def description(self):
        if self == LanguageEnums.ENGLISH:
            return "Used for defining if English is the preferred language"
        elif self == LanguageEnums.FRENCH:
            return "Used for defining if French is the preferred language"
        else:
            return "Error, this enum value does not exist."

File: api/src/enums/languages.js
import {GraphQLEnumType} from 'graphql'

export const LanguageEnums = new GraphQLEnumType({
  name: 'LanguageEnums',
  values: {
    ENGLISH: {
      value: 'english',
      description: 'Used for defining if English is the preferred language.',
    },
    FRENCH: {
      value: 'french',
      description: 'Used for defining if French is the preferred language.',
    },
  },
  description: "An enum used to define user's language.",
})
